Dockerfile

```diff
@@ -1,80 +1,118 @@
-
-# Additional runtime dependencies go here
-FROM ${BUILD_BASE} as base
+# Run-time dependencies go here
+FROM ${BUILD_BASE} AS base
 ARG UBUNTU_VERSION
+ARG LIBRARIES
 RUN apt-get update && \
-    cat /etc/lsb-release && \
-    if [ "${UBUNTU_VERSION}" = "20.04" ] ; then \
-        apt-get install -qqy --no-install-recommends libtinfo6 ; \
-    else \
-        apt-get install -qqy --no-install-recommends libtinfo5 ; \
-    fi && \
+    apt-get install -qqy --no-install-recommends python3 python3-pip python3.8 python3.8-venv python3-setuptools xz-utils && \
     rm -rf /var/lib/apt/lists/*
 
 # Build-time dependencies go here
-FROM trailofbits/cxx-common:llvm${LLVM_VERSION}-${DISTRO_BASE}-${ARCH} as deps
-
-ENV DEBIAN_FRONTEND=noninteractive
+FROM trailofbits/cxx-common-vcpkg-builder-ubuntu:${UBUNTU_VERSION} as deps
+ARG UBUNTU_VERSION
+ARG ARCH
+ARG LLVM_VERSION
+ARG LIBRARIES
 
 RUN apt-get update && \
-    apt-get install -qqy ninja-build python3.8 python3-pip python3.8-venv liblzma-dev zlib1g-dev libtinfo-dev curl git wget build-essential ninja-build ccache clang && \
+    apt-get install -qqy xz-utils python3.8-venv make rpm && \
     rm -rf /var/lib/apt/lists/*
 
-COPY --from=remill /opt/trailofbits /opt/trailofbits
+# Build dependencies
+WORKDIR /dependencies
+RUN git clone --depth=1 --branch master https://github.com/lifting-bits/remill.git && \
+    cd remill && \
+    ./scripts/build.sh --llvm-version ${LLVM_VERSION} --prefix ${LIBRARIES} --download-dir /dependencies
 
-# Source code build
-FROM deps as build
-ARG LIBRARIES
+# Make this a separate RUN because the build script above downloads a lot
+RUN cd remill && \
+    dpkg -i remill-build/*.deb
 
+# Source code build
+FROM deps AS build
 WORKDIR /anvill
-COPY . ./
+ARG UBUNTU_VERSION
+ARG ARCH
+ARG LLVM_VERSION
+ARG LIBRARIES
 
-ENV CC="/usr/bin/clang"
-ENV CXX="/usr/bin/clang++"
-ENV TRAILOFBITS_LIBRARIES="${LIBRARIES}"
 ENV VIRTUAL_ENV=/opt/trailofbits/venv
-ENV PATH="${VIRTUAL_ENV}/bin:${LIBRARIES}/llvm/bin:${LIBRARIES}/cmake/bin:${LIBRARIES}/protobuf/bin:${PATH}"
+ENV PATH="${VIRTUAL_ENV}/bin:${PATH}"
 
 # create a virtualenv in /opt/trailofbits/venv
 RUN python3.8 -m venv ${VIRTUAL_ENV}
 
-RUN mkdir -p build && cd build && \
-    cmake -G Ninja -DCMAKE_PREFIX_PATH=/opt/trailofbits/remill -DCMAKE_VERBOSE_MAKEFILE=True -DCMAKE_INSTALL_PREFIX=/opt/trailofbits/anvill .. && \
-    cmake --build . --target install
+# Needed for sourcing venv
+SHELL ["/bin/bash", "-c"]
+
+COPY . ./
+
+RUN source ${VIRTUAL_ENV}/bin/activate && \
+    cmake -G Ninja -B build -S . \
+        -DANVILL_ENABLE_INSTALL_TARGET=true \
+        -Dremill_DIR:PATH=${LIBRARIES}/lib/cmake/remill \
+        -DANVILL_INSTALL_PYTHON3_LIBS=ON \
+        -DCMAKE_INSTALL_PREFIX:PATH="${LIBRARIES}" \
+        -DCMAKE_VERBOSE_MAKEFILE=True \
+        -DVCPKG_ROOT=/dependencies/vcpkg_ubuntu-${UBUNTU_VERSION}_llvm-${LLVM_VERSION}_amd64 \
+        && \
+    cmake --build build --target install
```

Comment:
Once `ANVILL_INSTALL_PYTHON3_LIBS` is removed, both the native library and the Python module can be installed with:

```
cd build

mkdir destdir
export DESTDIR="$(pwd)/destdir"
anvill_version=$(git describe --always)

cmake --build . --target install
cpack -D ANVILL_DATA_PATH="${DESTDIR}" -R ${anvill_version} --config ../packaging/main.cmake

dpkg -i *.deb
```

Reply:
I think we may also be able to simplify this with `cpack -G DEB`; will take a look

Reply:
We also specifically want to install into a venv here, so the regular package installation process wont do it.

Reply:
Removed `ANVILL_INSTALL_PYTHON3_LIBS`

Reply:
Hah. Posted this comment before getting to this one https://github.com/lifting-bits/anvill/pull/197#discussion_r632073716

Yes, I like your idea @alessandrogario